ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java

```diff
@@ -22,7 +22,7 @@ Changes copyright (C) 2020 by Maksim Khramov
 
 /** This class presents a user interface for entering object scripts. */
 
-public class ScriptedObjectEditorWindow extends ScriptWindowBase
+public final class ScriptedObjectEditorWindow extends ScriptWindowBase
```

Comment:
Why the change to final?

Reply:
This way (Plus ScriptWindowBase is marked as package-private) I prevent unintended extension  this classes.

Reply:
Again, why?

If this is in response to the concern about the structure that I expressed the first time through, you seem to have misunderstood me...

Reply:
Just not found good solution with composition style against inheritance style

Reply:
Why do you want to prevent people from extending it?

The point was that if the editor is a widget, anyone can embed an editor inside any window.  But if the editor is a window, it can only be used in that window and no other.

Reply:
As @Sailsman63 wrote

> As you've written this, one has to extend the ScriptWindow

I close this views (Script editor and ScriptObject editor) from extending nd reduce visibility of their base class. 
... Maybe I do not understand the idea in general

Reply:
I can run a PR against your branch to show you what I mean. It may be a few days.